### DISCUSSION TITLE
Gradio function argument with multiple arguments

### DISCUSSION THREAD

**Discussion by longtongster** [mrdbourke/pytorch-deep-learning#459] [General]
Does anyone knows if the function required in the Interface can work with multiple arguments?

**Comment by mrdbourke** [comment]
Hi @longtongster,

You can typically use multiple inputs with Gradio via a list input.

For example:

```python
import gradio as gr

interface = gr.Interface(
 fn=function,
 inputs=[input_1, input_2, input_3],
 outputs=[output_1]
)
```

See more in this video: https://youtu.be/LIO-G7AVgBM

**Reply by longtongster** [reply]
Hi @mrdbourke 

Thanks a lot for your swift response. This is helpful to a certain extend. 

In my case I want to use not multiple inputs in the app but more use a predict function that has for example the model, class_names and device as input (predict(img, model, class_names, device). It is unclear how to use gradio in that case as the function now requires 4 inputs where only one (img) comes from the app. I found that we can use partial from the functools package.

`predict_ = partial(predict, model=effnetb2, class_names=class_names, device=device )`

creates a one argument function. This can be used in gradio app.

**Reply by bofcarbon1** [reply]
I used global variables and called them g_img, g_vit, g_vit_transforms and g_class_names. Then what start off as locals I just set to the global ones so I don't have to worry about passing them as function arguments/parameters. I took a modified  notebook  approach all in PyCharm though. I also chose ViT over EffNetB2 because the numbers favored ViT on a 'mps' GPU on a Mac Pro M2.
